### PR TITLE
openvpn: Allow override of interface name

### PR DESCRIPTION
**Pull request by miska**
If using a configuration file for OpenVPN, allow overriding name of the
interface. The reason is that then people could use configuration file
provided by VPN provider directly and override the name of the interface
to include it in correct firewall zone without need to alter the
configuration file.

Signed-off-by: Michal Hrusecky <michal@hrusecky.net>

### CODE REVIEW

File: package/network/services/openvpn/files/openvpn.init [hunk 1]
```diff
@@ -69,6 +69,38 @@ section_enabled() {
...
+	[ -n "$config_file" ] || return
+
+	# Check there is someething to override
+	config_get dev "$name" dev
+	config_get dev_type "$name" dev_type
+	[ -n "$dev" ] || return
+
+	# If there is a no dev_type, try to guess it
+	if [ -z "$dev_type" ]; then
+		if grep -q '^[[:blank:]]*dev-type[[:blank:]]*tun' "$conf" || \
```
Comment (jow-) [inline package/network/services/openvpn/files/openvpn.init:89]: Option values in native configuration may also be single or double quoted. Maybe we can reuse https://github.com/openwrt/openwrt/blob/master/package/network/services/openvpn/files/etc/hotplug.d/openvpn/01-user#L3 and use a logic similar to this one:

```
local dev dev_type
if get_option dev dev || get_option dev_type dev-type; then
    case "${dev_type:-$dev}" in
        tun*) echo "--dev-type tun --dev $dev" ;;
        ?*) echo "--dev-type tap --dev $tap" ;;
    esac
fi

**Comment by miska** [comment]
What about like this?

**Comment by jow-** [comment]
LGTM!

**Comment by jow-** [comment]
Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/jow.git

**Comment by jow-** [comment]
It is merged to master now.

**Pull request closed without merging**